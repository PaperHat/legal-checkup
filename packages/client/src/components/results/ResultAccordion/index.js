import * as React from 'react'
import Accordion from '../../UIKit/Accordion'
import ResultTopicHeader from '../ResultTopicHeader'
import ResultTopicContent from '../ResultTopicContent'

export default function ResultAccordion (props) {
  const { topicObj = {} } = props
  const { questions = [] } = topicObj

  const headingRender = (onClick) => {
    return <ResultTopicHeader name={topicObj.name} onClick={onClick} />
  }

  const contentRender = () => {
    return <ResultTopicContent questions={questions} />
  }

  return (
    <Accordion heading={headingRender} content={contentRender} isExpanded />
  )
}
