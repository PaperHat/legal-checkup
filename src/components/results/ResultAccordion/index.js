import * as React from 'react'
import { filter } from 'ramda'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import styledMap from 'styled-map'
import Accordion from '../../UIKit/Accordion'
import ResultTopicHeader from '../ResultTopicHeader'
import ResultTopicContent from '../ResultTopicContent'
import { DESKTOP, MOBILE, TABLET } from '../../../constants'
import { NO } from '../../../state/constants'

const Boundary = styled.div`
  background-color: #fff;
  margin: 15px auto;
  padding: ${styledMap('format', {
    [DESKTOP]: '23px 24px 21px 30px',
    [TABLET]: '16px 20px 12px 20px',
    [MOBILE]: '16px 20px 12px 20px'
  })};
  width: ${styledMap('format', {
    [DESKTOP]: '866px',
    [TABLET]: '275px',
    [MOBILE]: '275px'
  })};
`

export default function ResultAccordion (props) {
  const { format, topicName = {}, topicArray } = props
  const { questions = [] } = topicName

  const isExpanded = () => {
    if (topicArray.length > 2) {
      return false
    } else {
      return true
    }
  }
  const headingRender = onClick => {
    return (
      <ResultTopicHeader
        format={format}
        name={topicName.name}
        onClick={onClick}
        isExpanded={isExpanded}
      />
    )
  }

  const contentRender = () => {
    const showContent = question => question.answer !== NO
    const filteredQuestion = filter(showContent, questions)
    return <ResultTopicContent format={format} questions={filteredQuestion} />
  }
  return (
    <Boundary format={format}>
      <Accordion
        heading={headingRender}
        content={contentRender}
        isExpanded={isExpanded}
      />
    </Boundary>
  )
}

ResultAccordion.propTypes = {
  format: PropTypes.string.isRequired,
  topicName: PropTypes.object.isRequired,
  topicArray: PropTypes.array.isRequired
}
