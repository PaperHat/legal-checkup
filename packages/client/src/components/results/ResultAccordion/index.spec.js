import * as React from 'react'
import { create } from 'react-test-renderer'
import ResultAccordion from './'

describe('components:results:ResultAccordion', function () {
  const topic = {
    name: 'Money Troubles',
    questions: [
      {
        results: 'The fact that someone is chasing you for money',
        url: 'http://communitylaw.org.nz/community-law-manual/chapter-13-credit-and-debt/what-this-chapter-covers-chapter-13/'
      },
      {
        results: 'The problem you had recently had with a service you have brought',
        url: 'http://communitylaw.org.nz/community-law-manual/chapter-12-consumer-protection/what-this-chapter-covers-chapter-12/'
      }
    ]
  }

  it(`matches the snapshot`, () => {
    expect(create(<ResultAccordion topicObj={topic} />).toJSON()).toMatchSnapshot()
  })
})
