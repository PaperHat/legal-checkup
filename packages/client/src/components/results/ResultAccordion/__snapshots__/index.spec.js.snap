// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components:results:ResultAccordion matches the snapshot 1`] = `
<div>
  <div>
    <span>
      Money Troubles
    </span>
    <button
      onClick={[Function]}
    >
      show less/more
    </button>
  </div>
  <div>
    <p>
      <div>
        The fact that someone is chasing you for money
        ...
      </div>
      <a
        href="http://communitylaw.org.nz/community-law-manual/chapter-13-credit-and-debt/what-this-chapter-covers-chapter-13/"
      >
        What does the law say about this
      </a>
    </p>
    <p>
      <div>
        The problem you had recently had with a service you have brought
        ...
      </div>
      <a
        href="http://communitylaw.org.nz/community-law-manual/chapter-12-consumer-protection/what-this-chapter-covers-chapter-12/"
      >
        What does the law say about this
      </a>
    </p>
  </div>
</div>
`;
